esp-hal-common/src/usb_serial_jtag.rs

```diff
@@ -249,3 +266,122 @@ impl embedded_hal_nb::serial::Write for UsbSerialJtag<'_> {
         self.flush_tx_nb()
     }
 }
+
+#[cfg(feature = "async")]
+mod asynch {
+    use core::task::Poll;
+
+    use embassy_sync::waitqueue::AtomicWaker;
+    use procmacros::interrupt;
+
+    use super::{Error, Instance};
+    use crate::UsbSerialJtag;
+
+    // Single static instance of the waker
+    static WAKER: AtomicWaker = AtomicWaker::new();
+
+    pub(crate) struct UsbSerialJtagFuture<'d, T> {
```

Comment:
We can probably remove the generic parameter `T` here, since there's only ever one instance of `USB_DEVICE`.

Reply:
Done